src/components/Tokens/TokenTable/TokenTable.tsx

```diff
@@ -94,40 +88,39 @@ function useSortedTokens(addresses: string[], tokenData: Record<string, TokenDat
 
   return useMemo(
     () =>
-      addresses.sort((token1Address, token2Address) => {
+      tokenData &&
+      tokenData.sort((token1, token2) => {
         if (!tokenData) {
           return 0
         }
-        // fix any, delta property
-        const token1 = tokenData[token1Address] as any
-        const token2 = tokenData[token2Address] as any
-
+        // fix delta/percent change property
         if (!token1 || !token2 || !sortDirection || !sortCategory) {
           return 0
         }
         let a: number
         let b: number
         switch (sortCategory) {
           case Category.marketCap:
-            a = token1.marketCap
-            b = token2.marketCap
-            break
-          case Category.percentChange:
-            a = token1.delta
-            b = token2.delta
+            a = token1.marketCap?.value ?? 0
```

Comment:
not convinced defaulting to 0 is correct here?

Reply:
ya same, if it's not there, should we just put it at the bottom? this is for sort right

Reply:
yes